node/nodeTransactions_test.go

```diff
@@ -464,3 +466,52 @@ func setupGetMiniblockMetadataByTxHash(historyRepo *testscommon.HistoryRepositor
 		}, nil
 	}
 }
+
+func TestPrepareUnsignedTx(t *testing.T) {
+	t.Parallel()
+	addrSize := 32
+	scr1 := &smartContractResult.SmartContractResult{
+		Nonce:          1,
+		Value:          big.NewInt(2),
+		SndAddr:        bytes.Repeat([]byte{0}, addrSize),
+		RcvAddr:        bytes.Repeat([]byte{1}, addrSize),
+		OriginalSender: []byte("invalid original sender"),
+	}
+
+	n := Node{}
+	n.addressPubkeyConverter, _ = pubkeyConverter.NewBech32PubkeyConverter(addrSize)
+
+	scrResult1, err := n.prepareUnsignedTx(scr1)
+	assert.Nil(t, err)
+	expectedScr1 := &transaction.ApiTransactionResult{
+		Tx:       scr1,
+		Nonce:    1,
+		Type:     string(transaction.TxTypeUnsigned),
+		Value:    "2",
+		Receiver: "erd1qyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsl6e0p7",
+		Sender:   "erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu",
```

Comment:
maybe check explicity that the `OriginalSender` is nil/empty

Reply:
done